core/src/platform.cpp

```diff
@@ -79,8 +80,7 @@ UrlRequestHandle Platform::startUrlRequest(Url _url, UrlCallback&& _callback) {
         UrlResponse response;
         response.error = shutdown_message;
         _callback(std::move(response));
-
-        return UrlRequestNotCancelable;
+        return 0;
```

Comment:
`0` is still a special case, can we either document it or use a constexpr for it?

Reply:
I would suggest to use `constexpr` and use that at all the places below, where intended.

Reply:
It has the very special meaning: "Platform is shutting down - we have already handled the request synchronously. You don't need a handle since there is nothing left to handle". 

We could also return an incremented urlRequestCount but that would not change anything

Reply:
Ok the `_request ==  0` check in cancelUrlRequest is a bit superfluous..

Reply:
Hmm yeah maybe 0 should have the meaning `request_done` for synchronous handled requests, so that callers know not to add these handles into their queue or something